src/app/cluster/details/external-cluster/external-node-list/component.ts

```diff
@@ -72,11 +71,15 @@ export class ExternalNodeListComponent implements OnInit, OnChanges, OnDestroy {
   }
 
   isLoadingData(): boolean {
-    return (_.isEmpty(this.nodes) && !this.isClusterRunning) || !this.isInitialized;
+    return (_.isEmpty(this.nodes) && !this.isRunning()) || !this.isInitialized;
   }
 
   hasNoData(): boolean {
-    return _.isEmpty(this.nodes) && this.isClusterRunning && this.isInitialized;
+    return _.isEmpty(this.nodes) && this.isRunning() && this.isInitialized;
+  }
+
+  isRunning(): boolean {
```

Comment:
This is the same method which was bound with `isClusterRunning` Input so how does this make the difference?

Reply:
Correct! At list details page same logic is used to check state from cluster and pass `isClusterRunning` passing as input where in case of MD details page -> `isClusterRunning` is never passed -> which makes !isClusterRunning -> truthy values and satisfies loader condition therefore it just keep displaying loading state in MD details page where nodes are shown